### PR TITLE
Replace Summary with Histogram for all metrics

### PR DESCRIPTION
**Pull request by xiang90**
Histogram makes more sense for etcd runtime debugging. etcd is very sensitive for any high spike latency and we want to know the exact range of the spike. It is because a high spike might block raft routines and stop leader from sending heartbeat, which leads to unstable cluster.

etcd cares less about very accurate 99th percentile.

### CODE REVIEW

File: rafthttp/metrics.go [hunk 1]
```diff
@@ -23,12 +23,13 @@ import (
...
-	msgSentDuration = prometheus.NewSummaryVec(
-		prometheus.SummaryOpts{
+	msgSentDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
 			Namespace: "etcd",
 			Subsystem: "rafthttp",
-			Name:      "message_sent_latency_microseconds",
+			Name:      "message_sent_latency_seconds",
 			Help:      "message sent latency distributions.",
+			Buckets:   prometheus.ExponentialBuckets(0.0005, 2, 13),
```
Comment (yichengq) [inline rafthttp/metrics.go]: The snapshot send duration is also reported in msgSentDuration, and it may take more time.

Comment (xiang90) [inline rafthttp/metrics.go]: We should probably consider moving snapshot metrics out of this vector. It worth monitoring separately.

Comment (yichengq) [inline rafthttp/metrics.go]: agree. leaving a TODO is good to me.

**Comment by xiang90** [comment]
@yichengq PTAL

**Comment by yichengq** [comment]
LGTM

File: Documentation/metrics.md [hunk 2]
```diff
@@ -15,16 +15,16 @@ etcd now exposes the following metrics:
...
+|-----------------------------------------|--------------------------------------------------|-----------|
+| file_descriptors_used_total             | The total number of file descriptors used        | Gauge     |
+| proposal_durations_seconds              | The latency distributions of committing proposal | Histogram |
+| pending_proposal_total                  | The total number of pending proposals            | Gauge     |
+| proposal_failed_total                   | The total number of failed proposals             | Counter   |
 
 High file descriptors (`file_descriptors_used_total`) usage (near the file descriptors limitation of the process) indicates a potential out of file descriptors issue. That might cause etcd fails to create new WAL files and panics.
 
-[Proposal](glossary.md#proposal) durations (`proposal_durations_milliseconds`) give you an summary about the proposal commit latency. Latency can be introduced into this process by network and disk IO.
+[Proposal](glossary.md#proposal) durations (`proposal_durations_seconds`) give you an histogram about the proposal commit latency. Latency can be introduced into this process by network and disk IO.
```
Comment (jonboulle) [inline Documentation/metrics.md]: s/an histogram/a histogram/

**Pull request merged**